core/server/common/src/main/java/alluxio/master/BackupManager.java

```diff
@@ -106,8 +107,9 @@ public void backup(OutputStream os, AtomicLong entryCount) throws IOException {
     GzipCompressorOutputStream zipStream = new GzipCompressorOutputStream(os);
 
     // Executor for taking backup.
-    CompletionService<Boolean> completionService = new ExecutorCompletionService<>(
-        Executors.newFixedThreadPool(4, ThreadFactoryUtils.build("master-backup-%d", true)));
+    ExecutorService es = Executors.newFixedThreadPool(
+        2, ThreadFactoryUtils.build("master-backup-%d", true));
```

Comment:
why 4->2?

Reply:
There seem to be only 2 tasks ever submitted to the pool, but maybe I was reading it wrong since the test timed out?

Reply:
I changed it back to 4 and it works.